File: Data Structures & Algorithms/jump-game/submission-2.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<bool> vis(n, false);
        vis[0] = true;
        for(int i = 0; i < n; i++)
        {
            if(!vis[i]) continue;
            for(int j = 1; j <= nums[i] && i + j < n; j++)
            {
                vis[i + j] = true;
            }
        }
        return vis[n-1];
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-2.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = [-1] * len(nums)
        dp[0] = 0
        for i in range(len(nums)):
            if dp[i] == -1:
                continue
            for j in range(nums[i] + 1):
                if i + j >= len(nums):
                    continue
                if dp[i + j] == -1:
                    dp[i + j] = dp[i] + 1
                else:
                    dp[i + j] = min(dp[i + j], dp[i] + 1)
        return dp[len(nums) - 1]